haskell/src/Monocle/Backend/Queries.hs

```diff
@@ -327,20 +328,22 @@ data RepoSummary = RepoSummary
   }
   deriving (Show, Eq)
 
-getReposSummary :: [BH.Query] -> TenantM [RepoSummary]
-getReposSummary basequery = do
-  rlist <- getRepos basequery
+getReposSummary :: QueryM [RepoSummary]
+getReposSummary = do
+  basequery_ <- getQuery
+  let basequery = toBHQuery basequery_
+  rlist <- liftTenantM $ getRepos basequery
   let names = term <$> rlist
-  sequence $ getRepoSummary <$> names
+  sequence $ getRepoSummary basequery <$> names
   where
-    getRepoSummary fn = do
+    getRepoSummary basequery fn = do
       let query = mkAnd $ basequery <> getQueryFromSL ("repo: " <> fn)
       totalChanges' <- countEvent query "ChangeCreatedEvent"
       abandonedChanges' <- countEvent query "ChangeClosedEvent"
       mergedChanges' <- countEvent query "ChangeMergedEvent"
       let openChanges' = totalChanges' - (abandonedChanges' + mergedChanges')
       pure $ RepoSummary fn totalChanges' abandonedChanges' mergedChanges' openChanges'
-    countEvent query docType = countEvents (documentType [query] docType)
+    countEvent query docType = liftTenantM $ countEvents (documentType [query] docType)
```

Comment:
Since countEvent is only used in getRepoSummary, I would suggest to define it L341 without the query parameter. That would make the call site shorter, (e.g.: `totalChanges' <- countEvent "ChangeCreatedEvent"`)